packages/compose/src/hooks/use-instance-id/index.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+const instancesMap = new WeakMap();
+
+/**
+ * Creates a new instance id for a given function.
+ *
+ * @param {Function} fn Function to use the id for.
+ */
+function findId( fn ) {
+	const instances = instancesMap.get( fn ) || 0;
+	instancesMap.set( fn, instances + 1 );
+	return instances;
+}
+
+/**
+ * Provides a unique instance ID.
+ *
+ * @param {Function} fn Function to use the id for.
```

Comment:
I think this should be document as Object as it works for any object.

Reply:
It does work happen to work for any object, but is that something we should accept?

Reply:
I don't see why we won't. For me there's no direct relationship between the argument and the instance aside the fact that it serves as a "context"/"group". It could even be a string. Having it be an object is good though in order to use WeakMap

Reply:
Ok, done. :)